src/lidar_tools/dsm_functions.py

```diff
@@ -72,7 +72,160 @@ def tap_bounds(site_bounds: tuple | list | np.ndarray, res: int | float) -> list
         nearest_ceil(site_bounds[2], res),
         nearest_ceil(site_bounds[3], res),
     ]
+def execute_ept_3dep_pipeline(extent_polygon,target_wkt,output_prefix,
+                                raster_resolution=1.0,
+                                tile_size_km=1.0,
+                                buffer_value=5,
+                                process_specific_3dep_survey=None,process_all_intersecting_surveys=False):
+    gdf = gpd.read_file(extent_polygon)
+    
+    # specify the output CRS of DEMs
+    with open(target_wkt, "r") as f:
+        OUTPUT_CRS = " ".join(f.read().replace("\n", "").split())
+    # fetch the readers for the pointclouds
+    readers, POINTCLOUD_CRS, extents, original_extents = return_readers(
+        gdf,
+        pointcloud_resolution=1,
+        tile_size_km=tile_size_km,
+        buffer_value=buffer_value,
+        return_specific_3dep_survey=process_specific_3dep_survey,
+        return_all_intersecting_surveys=process_all_intersecting_surveys
+    )
+    
+    output_path = Path(output_prefix).parent
+    prefix = Path(output_prefix).name
+    output_path = Path(output_path)
+    output_path.mkdir(exist_ok=True)
+    
+    print(f"Number of readers: {len(readers)}")
+    dsm_pipeline_list = []
+    dsm_fn_list = []
+    dtm_pipeline_list = []
+    dtm_no_fill_fn_list = []
+    dtm_fill_fn_list = []
+    intensity_pipeline_list = []
+    intensity_fn_list = []
+    for i, reader in enumerate(readers):
```

Comment:
You can leave it for a later PR, but just noting there is a lot of repeated code in this for-loop, and it seems like a logical way to break it up is to maybe have a function `create_pipeline_list(product='dsm')` (or 'dtm' or 'intensity'). That said I haven't looked closely to see if there are special cases for each!